src/Lecture3.hs

```diff
@@ -159,10 +185,16 @@ monsters, you should get a combined treasure and not just the first
 🕯 HINT: You may need to add additional constraints to this instance
   declaration.
 -}
-instance Semigroup (Treasure a) where
+instance (Num a) => Semigroup (Treasure a) where
```

Comment:
You can generalize the constraint from `Num` to `Semigroup` here and use `<>` in the implementation.

```suggestion
instance (Semigroup a) => Semigroup (Treasure a) where
```